src/modules/heartbeat/heartbeat.c

```diff
@@ -0,0 +1,155 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* heartbeat.c - publish regular heartbeat messages
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <flux/core.h>
+
+#include <src/common/libutil/fsd.h>
+
+static const double default_period = 2.0;
+
+struct heartbeat {
+    flux_t *h;
+    double period;
+    flux_watcher_t *timer;
+    flux_msg_handler_t **handlers;
+    flux_future_t *f;
+};
+
+static void heartbeat_get_cb (flux_t *h,
+                              flux_msg_handler_t *mh,
+                              const flux_msg_t *msg,
+                              void *arg)
+{
+    struct heartbeat *hb = arg;
+
+    if (flux_request_decode (msg, NULL, NULL) < 0)
+        goto error;
+    if (flux_respond_pack (h,
+                           msg,
+                           "{s:f}",
+                           "period",
+                           hb->period) < 0)
+        flux_log_error (h, "error responding to heartbeat.get request");
+    return;
+error:
+    if (flux_respond_error (h, msg, errno, NULL) < 0)
+        flux_log_error (h, "error responding to heartbeat.get request");
+}
+
+static void timer_cb (flux_reactor_t *r,
+                      flux_watcher_t *w,
+                      int revents,
+                      void *arg)
+{
+    struct heartbeat *hb = arg;
+
+    flux_future_destroy (hb->f);
+    if (!(hb->f = flux_event_publish (hb->h, "heartbeat.pulse", 0, NULL)))
+        flux_log_error (hb->h, "error publishing heartbeat");
```

Comment:
Is `hb->f` really necessary? It seems like it isn't actually used in the module - could you instead use a local `flux_future_t` and destroy it just after creation? Sorry if I missed something obvious.

Reply:
That's a good point!  I was sort of half way between that and adding a continuation to capture any publishing errors.  I should probably just add the continuation.